### PR TITLE
Event based waiting

### LINKED ISSUE #85: Event based waiting
Currently, we use `time.sleep` to wait for an event to happen. Examples,

1. Waiting for testnet to complete its setup and ready to serve.
2. Waiting for a transaction to be included in the blockchain.

These should be handled by some event-based mechanism.

### PR DESCRIPTION
**Pull request by rnbguy**
Closes #85

### CODE REVIEW

**Comment by rnbguy** [comment]
With this, one can remove `time.sleep` calls from reactors as following.

```
@step("init")
def init(testnet):
    testnet.oneshot()

    with TmEventSubscribe({"tm.event": "NewBlock"}):
        pass

    print("Testnet is producing blocks")

@step("action")
def action(testnet):
    ...
    tx = granter_wallet.create_and_sign_tx(...)

    with TmEventSubscribe({"tm.event": "Tx"}) as f:
        result = lcdclient.tx.broadcast(tx)
        f.set_filter(lambda msg: msg.events.tx.hash[0] == result.txhash)
        
    print("action is executed")
```

File: atomkraft/chain/utils.py [hunk 1]
```diff
@@ -46,7 +46,11 @@ async def _wait(self):
         while True:
             msg = responses.to_result(json.loads(await self.websocket.recv()))
             if "query" in msg.result and msg.result["query"] == self.query:
-                break
+                try:
+                    if self.filter(msg.result):
```
Comment (ivan-gavran) [inline atomkraft/chain/utils.py]: I am a bit confused by this line: where is the member function `filter` defined?

**Review by ivan-gavran — CHANGES_REQUESTED** [review]
It feels like a useful work, but somewhat rushed up. In particular, I would like to see:
 - update to `atomkraft/examples/...` to reflect the changes in this PR
 - some tests of the functionality

File: atomkraft/chain/utils.py (hunk 1 above)
Comment (rnbguy) [reply atomkraft/chain/utils.py]: I added a `set_filter` method.

**Review by andrey-kuprianov — APPROVED** [review]
Looks good, Rano!

**Pull request merged**